piker/clearing/_ems.py

```diff
@@ -638,66 +674,181 @@ async def _emsd_main(
+            # when also pulling data from feeds
+            open_trades_endpoint = paper.open_paperboi(
+                broker=broker,
+                symbol=symbol,
+                loglevel=loglevel,
+            )
+
+        else:
+            # open live brokerd trades endpoint
+            open_trades_endpoint = portal.open_context(
+                trades_endpoint,
+                loglevel=loglevel,
+            )
+
+        async with (
+            open_trades_endpoint as (brokerd_ctx, positions),
+            brokerd_ctx.open_stream() as brokerd_trades_stream,
+        ):
+            # signal to client that we're started
+            # TODO: we could eventually send back **all** brokerd
+            # positions here?
+            await ems_ctx.started(positions)
+
+            # establish 2-way stream with requesting order-client and
+            # begin handling inbound order requests and updates
+            async with ems_ctx.open_stream() as ems_client_order_stream:
+
+                # trigger scan and exec loop
+                n.start_soon(
+                    clear_dark_triggers,
+
+                    brokerd_trades_stream,
+                    ems_client_order_stream,
+                    feed.stream,
+
+                    broker,
+                    symbol,
+                    book
+                )
 
-                    await n.start(
-                        process_broker_trades,
-                        ctx,
-                        feed,
-                        dark_book,
-                    )
+                # begin processing order events from the target brokerd backend
+                # by receiving order submission response messages,
+                # normalizing them to EMS messages and relaying back to
+                # the piker order client.
+                n.start_soon(
+                    translate_and_relay_brokerd_events,
+
+                    broker,
+                    ems_client_order_stream,
+                    brokerd_trades_stream,
+                    dark_book,
+                )
 
-                    # start inbound order request processing
-                    await process_order_cmds(
-                        ctx,
-                        order_stream,
-                        symbol,
-                        feed,
-                        client,
-                        dark_book,
-                    )
+                # start inbound (from attached client) order request processing
+                await process_client_order_cmds(
+                    ems_client_order_stream,
+                    brokerd_trades_stream,
+                    symbol,
+                    feed,
+                    dark_book,
+                )
+
+
+class _Router(BaseModel):
+    '''Order router which manages per-broker dark books, alerts,
+    and clearing related data feed management.
+
+    '''
+    nursery: trio.Nursery
+
+    feeds: dict[str, tuple[trio.CancelScope, float]] = {}
```

Comment:
don't think we're stashing cancel scopes any more, so this type annot needs a fix.